### ISSUE TITLE
Docs: Mention that datadog metric tags replace default_tags

### ISSUE THREAD

**Issue by ecokes** [awslabs/aws-lambda-powertools-python#2990]
### What were you searching in the docs?

First off, I've been enjoying this library and I think the docs are pretty great overall. 

I did have one point of confusion I wanted to clarify. I've been using the DatadogMetrics class, (using this [section of the docs](https://docs.powertools.aws.dev/lambda/python/latest/core/metrics/datadog/#creating-metrics)) to add tags to metrics. I was surprised to find that adding tags in the `add_metric` method like this:
```
metrics.add_metric(name="SuccessfulBooking", value=1, tag2="bar", tag3="baz")
```
results in any default tags that were added to the metrics class, like this:
```
metrics.set_default_tags(tag1="foo")
```
getting removed from the metric. I assumed that default tags and additional one off tags would be combined in that case, so that the metric would end up as:
```
{
    "m": "SuccessfulBooking",
    "v": 1,
    "e": 1692736997,
    "t": [
        "tag1:foo",
        "tag2:bar",
        "tag3:baz"
    ]
}
```

However, the metric instead comes out as:
```
{
    "m": "SuccessfulBooking",
    "v": 1,
    "e": 1692736997,
    "t": [
        "tag2:bar",
        "tag3:baz"
    ]
}
```

I assumed I had configured something wrong for awhile, until I found this [section of the code](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/aws_lambda_powertools/metrics/provider/datadog/datadog.py#L330). So it looks like that's the intended behavior, but it would be helpful to add a line in the docs identifying that default_tags and metric tags are one or the other, not additive.

### Is this related to an existing documentation section?

https://docs.powertools.aws.dev/lambda/python/latest/core/metrics/datadog/#adding-tags

### How can we improve?

Mention that metric tags overwrite default tags instead of being appended.

### Acknowledgment

- [X] I understand the final update might be different from my proposed suggestion, or refused.

**Comment by heitorlessa** [comment]
@leandrodamascena i thought they are additive too?

except when they're not unique (explicit override vs default)

@ecokes let's wait for Leandro. I suspect this is a bug in the example you provided but maybe Leandro or @roger-zhangg identified a problem I'm unaware

**Comment by heitorlessa** [comment]
It’s a bug!! Thank you so much for reporting it!!

Line 330 — we’ll need to filter for duplicates first and use precedence,
and it’s also missing a functional test for this case

https://github.com/aws-powertools/powertools-lambda-python/blob/develop/aws_lambda_powertools/metrics/provider/datadog/datadog.py

**Comment by ecokes** [comment]
@heitorlessa Thanks for taking a look, I was wondering if it might be a bug myself.

**Comment by leandrodamascena** [comment]
Hello @ecokes! Thanks for reporting this. I'm working on submitting a fix for this.

We usually release new versions on Friday, but if it's too critical for you at this moment, we can release a patch release tomorrow. Please let me know what you prefer :)

Thanks

**Comment by ecokes** [comment]
@leandrodamascena, Friday is fine with me, thanks for fixing it so quickly!

**Comment by leandrodamascena** [comment]
Hei @ecokes! We fixed the problem in the PR #2997.
